noir-projects/aztec-nr/aztec/src/context/public_context.nr

```diff
@@ -197,6 +200,24 @@ impl ContextInterface for PublicContext {
         self.new_nullifiers.push(side_effect);
         self.side_effect_counter = self.side_effect_counter + 1;
     }
+
+    fn nullifier_exists(self, nullifier: Field) -> bool {
+        let header = self.get_header();
+        let not_in = nullifier_valid_inclusion(nullifier, header); 
+        let is_in = nullifier_valid_non_inclusion(nullifier, header);
+        // We want to ensure that the user could not just make the nullifier inclusion 
+        // fail with a bad path, therefore it must either be in or not
+        let xor = (not_in | is_in) - (not_in & is_in);
+        xor == false
+    }
+
+    fn nullifier_valid_inclusion(self, nullifier: Field) -> bool {
```

Comment:
As above, I'm ok with just having the `exists` in the context, and then people can go to libraries for stuff that can more easily be messed up if they want to.